### ISSUE TITLE
Linking to ffmpeg/4.0 causes warnings

### ISSUE THREAD

**Issue by madebr** [bincrafters/community#423]
### Description of Problem, Request, or Question
When linking to `ffmpeg/4.0@bincrafters/stable`, some warnings are displayed.
`LNK4098` worries me.
```
  LINK : warning LNK4098: defaultlib 'LIBCMT' conflicts with use of other libs; use /NODEFAULTLIB:library [C:\issues\conan_ffmpeg_1\build\issue.vcxproj]
  libavcodec.lib(libx264.o) : warning LNK4049: locally defined symbol x264_levels imported [C:\issues\conan_ffmpeg_1\build\issue.vcxproj]
  libavcodec.lib(libx264.o) : warning LNK4049: locally defined symbol x264_bit_depth imported [C:\issues\conan_ffmpeg_1\build\issue.vcxproj]
```
Also: The log contain this line
```
  vpx.lib(vpx_src_vpx_image.obj) : MSIL .netmodule or module compiled with /GL found; restarting link with /LTCG; add /LTCG to the link command line to improve linker performance
```
Is lto by default really required?
### Package Details (Include if Applicable)
* Package Name/Version: **ffmpeg/4.0**
* Operating System: **Windows 10**
* Compiler+version: **Visual Studio 15 2017**

### Steps to reproduce (Include if Applicable)
reproducer: https://github.com/madebr/issues/tree/master/conan_ffmpeg_1
ci appveyor output: https://ci.appveyor.com/project/madebr/issues/build/1.0.129#L310

**Comment by SSE4** [comment]
LNK4049 seems to be an upstream issue, as **libx264.c** unconditionally defines **X264_API_IMPORTS** :
https://github.com/FFmpeg/FFmpeg/blob/master/libavcodec/libx264.c#L32
used by:
https://github.com/mstorsjo/x264/blob/master/x264.h#L51
this is obviously redundant in case of static build

**Comment by madebr** [comment]
The `/GL` flag is causes by this line:
```
https://github.com/webmproject/libvpx/blob/d748cfdad2378fc488fb5df4411adf5f2d5b99c9/build/make/gen_msvs_vcxproj.sh#L355
```

**Comment by madebr** [comment]
I believe `LNK4098` is caused by `conan-intel_media_sdk`?
See these lines: https://github.com/bincrafters/conan-intel_media_sdk/blob/67baf438f72e6b9f87858d02a5778137667290b2/conanfile.py#L52-L56 .
The `exelinkflags_intel_media_sdk` section of the generated `conanbuildinfo.txt` is empty.

Removing `if int(str(self.settings.compiler.version)) < 14:` causes `-nodefaultlib:libcmt` to be added to `exelinkflags` and makes the `LNK4098` warning disappear.

**Comment by SSE4** [comment]
yep, that's right
I'll try to update to 2018r1 this week

**Comment by madebr** [comment]
About the `/GL` and `/LTCG` flag,
adding
```
    def package_info(self):
        self.cpp_info.libs = ['vpx']
        if self.settings.os == 'Windows':
            if self.settings.build_type == 'Release':
                self.cpp_info.exelinkflags.append('-LTCG')
                self.cpp_info.sharedlinkflags.append('-LTCG')
```
to conan-libvpx's conanfile.py does not fix the warning but creates a new one:
```
Warning	LNK4075	ignoring '/INCREMENTAL' due to '/LTCG' specification	issue	C:\Users\maarten\Documents\Visual Studio 2017\Projects\issues\conan_ffmpeg_1\build\LINK	1	
```
So I guess it is better to remove the `WholeProgramOptimization` of libvpx.

**Comment by madebr** [comment]
I've opened a ticket https://trac.ffmpeg.org/ticket/7396

**Comment by madebr** [comment]
@SSE4 Am I correct to say that passing the correct flags to `configure` will make ffmpeg link to the correct x264 library?
https://trac.ffmpeg.org/ticket/7396#comment:3

**Comment by SSE4** [comment]
yes, that's correct
seems like ffmpeg developers don't case, so we'll have to patch sources to avoid warnings

**Comment by SSE4** [comment]
libvpx has weird issues on appveyor (not finding yasm), I am looking into

**Comment by SSE4** [comment]
status update:
1. added `intel_media_sdk/2018R2@bincrafters/stable` and changed package to avoid warning
2. libvpx is blocked on conan issue, waiting for conan 1.8

**Comment by SSE4** [comment]
status update:
libvpx resurrected with conan 1.8, so no longer blocked
I guess ffmpeg/4.0.2 build will be ready tomorrow with warnings fixed

**Comment by SSE4** [comment]
@madebr please try `ffmpeg/4.0.2@bincrafters/stable` package, it should have mentioned warnings fixed

**Comment by SSE4** [comment]
all warnings were fixed, closing
please open new issue if you see any other warnings
